src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -57,27 +58,67 @@ private HttpResponse ReadResponse(Stream responseStream)
             const int statusCodeStart = 9;
             const int statusCodeEnd = 12;
             const int startOfReasonPhrase = 13;
+            const int bufferSize = 10;
 
             // TODO: Get this from StringBuilderCache after we determine safe maximum capacity
             var stringBuilder = new StringBuilder();
 
-            var chArray = new byte[1];
-            void GoNextChar()
+            var chArray = new byte[bufferSize];
+
+            async Task GoNextChar()
             {
-                streamPosition++;
-                chArray[0] = (byte)responseStream.ReadByte();
+                await responseStream.ReadAsync(chArray, offset: 0, count: 1).ConfigureAwait(false);
                 currentChar = Encoding.ASCII.GetChars(chArray)[0];
+                streamPosition++;
+            }
+
+            async Task SkipUntil(int requiredStreamPosition)
+            {
+                var advanceCount = requiredStreamPosition - streamPosition;
+                // Not required in release mode, as should only arise from programming error
+                System.Diagnostics.Debug.Assert(advanceCount > 0, "RequiredStreamPosition should be greater than 0");
+                System.Diagnostics.Debug.Assert(advanceCount <= bufferSize, "RequiredStreamPosition should be less than buffer size");
+
+                await responseStream.ReadAsync(chArray, offset: 0, count: advanceCount).ConfigureAwait(false);
```

Comment:
`ReadAsync()` will return the number of bytes read, which can be less than the count requested. I think we may need to loop here until all the bytes we want are read.

Reply:
Good catch 👍

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1218/commits/656a470756c5926b3d68df33e3bfa4313e44babc. This throws is we unexpectedly reach the end of the stream. This won't actually occur with a NetworkStream, as either it will spin waiting for new data, or the connection will be closed and throw, but makes more sense logically.